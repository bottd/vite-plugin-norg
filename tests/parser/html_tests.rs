use vite_plugin_norg_parser::{extract_toc, transform};

#[test]
fn test_convert_nodes() {
    use rust_norg::NorgAST::Heading;

    let node = Heading {
        level: 1,
        title: vec![],
        content: vec![],
        extensions: Default::default(),
    };

    let ast = vec![node];
    let html = transform(&ast);
    let toc = extract_toc(&ast);
    assert!(html.starts_with("<h1 id=\"\">"));
    assert!(html.ends_with("</h1>"));
    assert_eq!(toc.len(), 1);
}
