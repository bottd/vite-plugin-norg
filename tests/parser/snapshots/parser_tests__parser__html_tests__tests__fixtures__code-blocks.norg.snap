---
source: tests/parser/html_tests.rs
expression: "(html, toc, metadata)"
---
- "<h1 id=\"code-blocks\">Code Blocks</h1>\n<h2 id=\"javascript\">JavaScript</h2>\n<pre class=\"lang-javascript\"><code class=\"lang-javascript\">function hello() { return &quot;world&quot;; }\n</code></pre><h2 id=\"python\">Python </h2>\n<pre class=\"lang-python\"><code class=\"lang-python\">def add(a, b): return a + b\n</code></pre><h2 id=\"plain\">Plain</h2>\n<pre><code>plain text\n</code></pre>"
- - level: 1
    title: Code Blocks
    id: code-blocks
  - level: 2
    title: JavaScript
    id: javascript
  - level: 2
    title: "Python "
    id: python
  - level: 2
    title: Plain
    id: plain
- title: "\"Code Test\""
